### PR TITLE
executor: insert negative primary key into auto_random table won't trigger rebase (#15397)

### PR DESCRIPTION
**Pull request by tangenta**
Cherry-pick #15397.

### Release note 

- Insert negative primary key into auto_random table won't trigger rebase.

### CODE REVIEW

File: executor/insert_common.go [hunk 1]
```diff
@@ -904,6 +904,9 @@ func (e *InsertValues) allocAutoRandomID(fieldType *types.FieldType) (int64, err
 }
 
 func (e *InsertValues) rebaseAutoRandomID(recordID int64, fieldType *types.FieldType) error {
+	if recordID < 0 {
+		return nil
+	}
```
Comment (AilinKid) [inline executor/insert_common.go:909]: rebase in allocator seems do the same thing (rebase < cur base return nil), so here for returning quickly?

Comment (tangenta) [reply executor/insert_common.go:909]: `auto_random` only rebase the incremental bits(lower bits). It is always positive, we prevent the case like `recordID < 0 < currentID < lower_bits(recordID)`.

Comment (AilinKid) [reply executor/insert_common.go:909]: got

**Comment by tangenta** [comment]
/run-unit-test
```
FAIL: sort_test.go:26: testSuite.TestSortInDisk

sort_test.go:60:
    c.Assert(tk.Se.GetSessionVars().StmtCtx.DiskTracker.MaxConsumed(), Greater, int64(0))
... compare_one int64 = 0
... compare_two int64 = 0
```

**Pull request merged**